Assets/MRTK/SDK/Experimental/Features/UX/BoundsControl/Visuals/BoxDisplay.cs

```diff
@@ -17,25 +17,36 @@ public class BoxDisplay
 
         private BoxDisplayConfiguration config;
 
+        private Vector3 cachedExtents = Vector3.zero;
+        private FlattenModeType cachedFlattenMode;
+
+        private bool isVisible = true;
+
         internal BoxDisplay(BoxDisplayConfiguration configuration)
         {
             Debug.Assert(configuration != null, "Can't create BoundsControlBoxDisplay without valid configuration");
             config = configuration;
+            config.flattenAxisScaleChanged.AddListener(UpdateDisplayWithCache);
+            config.materialChanged.AddListener(UpdateBoxDisplayMaterial);
         }
 
-        internal void AddBoxDisplay(Transform parent, Vector3 currentBoundsExtents, FlattenModeType flattenAxis)
+        ~BoxDisplay()
```

Comment:
because we don't have a lot of classes that are not components - usually people plan everything as component - this one though should just ecapsulate certain functionality that is internal to bounds control and should not be used as a standalone component - hence i've kept it as a regular class